packages/diagram-ui/src/features/move/movedElementsSelector.ts

```diff
@@ -0,0 +1,225 @@
+     */
+    private isMoved(elementId: string): boolean {
+        const fromLookup = this.isMovedLookup.get(elementId);
+        if (fromLookup != undefined) {
+            return fromLookup;
+        }
+        const element = this.index.getById(elementId);
+        if (element instanceof SCanvasElement || element instanceof SCanvasPoint) {
+            const isMoved = this.isElementMoved(element);
+            this.isMovedLookup.set(elementId, isMoved);
+            return isMoved;
+        } else {
+            throw new Error("Invalid element (should not be reachable)");
+        }
+    }
+
+    /**
+     * Checks if an element is moved.
+     *
+     * @param element the element to check
+     * @returns true if the element is moved, otherwise false
+     */
+    private isElementMoved(element: SCanvasElement | SCanvasPoint): boolean {
+        if (this.movedElements.has(element)) {
+            return true;
+        }
+        return this.isElementImplicitelyMoved(element);
+    }
+
+    /**
+     * Checks if an element is implicitely moved.
+     * - A canvas element is moved if its position is moved, if it is moved explicitly, or if the parent canvas is moved.
+     * - A relative point is moved if the target is moved.
+     * - A line point is moved if the (relevant segment of the) line provider is moved.
+     * - An absolute point is moved if the parent canvas is moved.
+     *
+     * @param element the element to check
+     * @returns true if the element is implicitely moved, otherwise false
+     */
+    private isElementImplicitelyMoved(element: SCanvasElement | SCanvasPoint): boolean {
+        if (element instanceof SCanvasElement) {
+            if (element.pos != undefined) {
+                return this.isMoved(element.pos);
+            }
+            return this.isParentCanvasImplicitelyMoved(element);
+        } else if (element instanceof SRelativePoint) {
+            return this.isMoved(element.target);
+        } else if (element instanceof SLinePoint) {
+            const lineProviderId = element.lineProvider;
+            const lineProvider = this.index.getById(lineProviderId) as SCanvasContent;
+            if (lineProvider instanceof SCanvasElement) {
+                return this.isMoved(lineProviderId);
+            }
+            if (lineProvider instanceof SCanvasConnection) {
+                const affectedSegment = LinePoint.calcSegmentIndex(element.pos, lineProvider.segments.length);
+                return this.isCanvasConnectionSegmentMoved(lineProvider, affectedSegment);
+            }
+        } else if (element instanceof SAbsolutePoint) {
+            return this.isParentCanvasImplicitelyMoved(element);
+        }
+        return false;
+    }
+
+    /**
+     * Checks if a canvas connection segment is moved
+     * It is moved if all relevant points (start, end, control points) are moved.
+     * If only some points are moved, it is considered a conflict, for which the hasConflict flag is set.
```

Comment:
> What happens then

That's up to the decision of the caller. It could ignore the issue, but most just not doing what it wants to do might be the better approach. Currently, in such a case the graphical edit is disabled (you can't drag anything with a selection which has a conflict)

> How can this occur

E.g. by selecting a line point which is relative to the first segment of a canvas connection, and then also the start point of said segment, but not the end point